modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2828,6 +2828,62 @@ def setitem_builder(df, internal_indices=[]):  # pragma: no cover
     # Drop/Dropna
     # This will change the shape of the resulting data.
     def dropna(self, **kwargs):
+        is_column_wise = kwargs.get("axis", 0) == 1
+        no_thresh_passed = kwargs.get("thresh", no_default) in (
+            no_default,
+            None,
+        )
+        # FIXME: this is a naive workaround for this problem: https://github.com/modin-project/modin/issues/5394
+        # if there are too many partitions then all non-full-axis implementations start acting very badly.
+        # The here threshold is pretty random though it works fine on simple scenarios
+        processable_amount_of_partitions = (
+            np.prod(self._modin_frame._partitions.shape) < CpuCount.get() * 32
+        )
```

Comment:
Let's not mix abstraction levels and separate this condition into a dataframe method.

Reply:
> Let's separate this condition into a dataframe method.

well, that would be quite hard...

here I want to dispatch between two different implementations of `.dropna()`. Since dataframe level doesn't define `dropna`, then I have to do this dispatching at the QC level.

The idea of a "general dispatcher" at the dataframe level was discussed at #5394 but trying suggested ideas didn't really find success, so there's a lot of work to be done in this regard

Reply:
Except for the `.processable_amount_of_partitions_for_dropna` field, I simply added a property `.num_parts` and kept the deciding logic at the QC, so the line now looks like:
```python
processable_amount_of_partitions = self._modin_frame.num_parts < CpuCount.get() * 32
```
It feels wrong adding the `.processable_amount_of_partitions_for_dropna` field, as the `modin_frame` layer should know nothing about the `dropna` and its implementation desires in terms of partitions amount